### PR TITLE
override nginx listening port and gunicorn server

### PR DESCRIPTION
**Pull request by lennon310**
## What changes are proposed in this pull request?
 
Make Nginx listening port and gunicorn server overridable. So every time the Mlflow Model image is ran, ENV can be set to change these two configurations. 
 
## How is this patch tested?
 
Unit test passed.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
The Nginx listening port and gunicorn server can be dynamically configured at model image runtime.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [x] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/container/__init__.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def _serve_pyfunc(model):
             _install_pyfunc_deps(MODEL_PATH, install_mlflow=True)
         bash_cmds += ["source /miniconda/bin/activate custom_env"]
     nginx_conf = resource_filename(mlflow.models.__name__, "container/scoring_server/nginx.conf")
```
Comment (ahutterTA) [inline mlflow/models/container/__init__.py]:
```
nginx_conf_template = resource_filename(mlflow.models.__name__, "container/scoring_server/nginx.conf.template")
nginx = Popen(['envsubst', '<', nginx_conf_template, '> <path to conf file>', '&&', 'nginx', '-c', '<path to conf file>'])

**Comment by mateiz** [comment]
Since these are running in Docker containers, why not use Docker's existing ability to map the container ports to other ports on the local machine (https://docs.docker.com/config/containers/container-networking/)? Do you want two models to be served from the same container somehow?

**Comment by lennon310** [comment]
Thanks @mateiz for the suggestion. That could be possible by setting V1ContainerPort during deployment, I will try whether it works for our use case and get back to you.

**Comment by lennon310** [comment]
@mateiz It turns out our company has some security policies that we are not allowed to set network host (ip, port) in containers, as a result, we could not map the port at runtime. We may still have to override the environment variables in order to reset the listening port and upstream server for each container in the pod. Please review. 
Thanks

**Pull request closed without merging**